sql/expression/procedureparam.go

```diff
@@ -90,29 +99,87 @@ func (ppr *ProcedureParamReference) Set(name string, val interface{}, valType sq
 }
 
 // HasBeenSet returns whether the parameter has had its value altered from the initial value.
-func (ppr *ProcedureParamReference) HasBeenSet(name string) bool {
-	name = strings.ToLower(name)
-	paramRefVal, ok := ppr.nameToParam[name]
-	if !ok {
+func (ppr *ProcedureParamReference) HasBeenSet(id int) bool {
+	if id >= len(ppr.idToParam) {
 		return false
 	}
-	return paramRefVal.HasBeenSet
+	return ppr.idToParam[id].HasBeenSet
 }
 
-func NewProcedureParamReference() *ProcedureParamReference {
-	return &ProcedureParamReference{make(map[string]*procedureParamReferenceValue)}
+// Open sets the designated cursor to open.
+func (ppr *ProcedureParamReference) Open(ctx *sql.Context, id int, name string, row sql.Row) error {
```

Comment:
More signs this type is doing too much

Reply:
This should be the responsibility of a Cursor object, that I previously got with `Get`, no?